src/main/java/org/jetbrains/kotlinx/lincheck/strategy/managed/ObjectManager.kt

```diff
@@ -0,0 +1,75 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.strategy.managed
+
+import java.util.*
+
+/**
+ * Object manager holds information about some object characteristics.
+ * Tracks the creation of local objects and leaks of their references to non-local objects.
+ */
+internal class ObjectManager {
+    // for each local object store all objects that depend on it (e.g, are referenced by it)
+    private val localObjects = IdentityHashMap<Any, MutableList<Any>>()
+    // some objects can have a name associated to them
```

Comment:
In general case, it can be sometimes convenient to have labels for some objects, so that to use them in method call representation `objectLabel.call(parameter): result`.
Yet only AFUs have such labels (their names are the names of the fields they are associated with), but I feel it is a good idea to make the generalization here.